File: api/api.web/Model/SnapshotWebModel.fs
﻿namespace Djambi.Api.Web.Model

open System.ComponentModel.DataAnnotations

type SnapshotInfoDto = {
    [<Required>]
    id : int

    [<Required>]
    createdBy : CreationSourceDto

    [<Required>]
    description : string
}

[<CLIMutable>]
type CreateSnapshotRequestDto = {
    [<Required>]
    [<StringLength(100, MinimumLength = 1)>]
    description : string
}

File: api/api.web.model/SnapshotWebModel.cs
namespace Apex.Api.Web.Model
{
    public class SnapshotInfoDto
    {
        public int Id { get; set; }
        public CreationSourceDto CreatedBy { get; set; }
        public string Description { get; set; }
    }

    public class CreateSnapshotRequestDto
    {
        public string Description { get; set; }
    }
}
